File: Day 8/268. Missing Number.java
class Solution {
    public int missingNumber(int[] nums) {
        int a =0;
        for (int i = 0; i<nums.length;i++){
            a ^=nums[i];
        }
        for (int i = 0; i<=nums.length;i++){
            a ^=i;
        }
        return a;
    }
}

File: Day 008/268. Missing Number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int t = 1;
        
        if (nums[0] != 0) {
            return 0;
        }
        else {
            while (t < nums.size() && (nums[t] - nums[t - 1]) == 1) {
                t++;
            }
            return nums[t - 1] + 1;
        }
    }
};

